validate.go

```diff
@@ -71,3 +71,14 @@ func ComposeValidators(validators ...Validator) Validator {
 		return nil
 	}
 }
+
+// isZero returns true if the passed value is the zero object
+func isZero(v reflect.Value) bool {
+	switch v.Kind() {
+	case reflect.Slice:
```

Comment:
To be pedantic, I think this should probably be:
```
case reflect.Slice, reflect.Map:
```
It turns out neither of these are considered zero values:
```
x := map[string]string{}
x := []string{}
```
So you should check the `Len` on both of them.  Slightly curiously the array version of this is fine, so `x := [4]string{}` is considered a zero value.